### PR TITLE
Close websocket connection after being hidden for 5 minutes

### LINKED ISSUE #5328: websocket - does not manage connections?
## Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

## The problem

There's an issue linked to issue https://github.com/home-assistant/core/issues/23938 - when some browser is left unrefreshed (at least what I've encountered, but I know someone who doesn't have unfrefreshed page and have same issue), there is tons of messages in LOG appearing:
`ERROR (MainThread) [homeassistant.components.websocket_api.http.connection.xxxx] Client exceeded max pending messages [2]: 512`
making high CPU usage instantly.
I've tried to debug why is that happening and:
- restart of HA
- reboot of machine
doesn't help.

I've also added websocket sensor (according to [docs](https://www.home-assistant.io/integrations/websocket_api/#track-current-connections)) to see how many connection HA has to handle. The sensor graph looks like this:
image
The spike going up is the moment when problem was shown in logs for around 30 minutes - the number of connections went straight up to ~180.
After I've restarted last machine with browser/HA opened, the number of connections didn't go up.
But neither it went down... It seems like HA/websocket is not handling those connection that doesn't do anything or are left somehow unmanaged.

## Expected behavior

Close connections that are not served as they should or are not needed anymore.
They didn't disappear after closing of browser that caused problems.

## Steps to reproduce

Open HA on browser and left it unattended for longer, while having i.e. graphs that updates themselves in regular intervals.

## Environment

arch | x86_64
-- | --
dev | false
docker | true
hassio | true
os_name | Linux
os_version | 4.19.0-8-amd64
python_version | 3.7.7
timezone | Europe/Warsaw
version | 0.107.5
virtualenv | false

- Home Assistant release with the issue: 0.107.5 (latest at the moment)
- Last working Home Assistant release (if known): have seen such errors previously
- Browser and browser version: Chrome
- Operating system: Windows (browser) / Debian (HA)

## Problem-relevant configuration

n/a

## Javascript errors shown in your browser console/inspector

n/a, didn't catch

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Needs https://github.com/home-assistant/home-assistant-js-websocket/pull/120

We will close the WebSocket connection to Home Assistant 5 minutes after the document has been hidden (change of tab, minimize, lock phone, etc).

We will reconnect when the document gets unhidden again.

Might fix https://github.com/home-assistant/frontend/issues/5328
Related https://github.com/home-assistant/android/pull/545

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/profile/ha-refresh-tokens-card.js [hunk 1]
```diff
@@ -96,6 +96,9 @@ class HaRefreshTokens extends LocalizeMixin(EventsMixin(PolymerElement)) {
 
   async _handleDelete(ev) {
     const token = ev.model.item;
+    if (token.is_current) {
```
Comment (balloob) [inline src/panels/profile/ha-refresh-tokens-card.js:99]: Should we just not show the delete icon for current?

File: src/state/connection-mixin.ts [hunk 2]
```diff
@@ -26,6 +26,10 @@ export const connectionMixin = <T extends Constructor<HassBaseEl>>(
   superClass: T
 ) =>
   class extends superClass {
+    private __hiddenTimeout?: number;
```
Comment (balloob) [inline src/state/connection-mixin.ts:29]: We shouldn't put this on the `connection-mixin`. This is re-used by different projects. Instead let's either add it to `home-assistant.ts` or a new mixin

File: src/panels/profile/ha-refresh-tokens-card.js (hunk 1 above)
Comment (bramkragten) [reply src/panels/profile/ha-refresh-tokens-card.js:99]: A disabled button shouldn't fire the click event, have to check it, shouldn't be in this PR.

**Review by balloob — APPROVED** [review]
🎉

**Comment by AmirHosseinKarimi** [comment]
Hi, HA will not shown new persist notification as system notification?

**Comment by andriej** [comment]
Will it be in 0.111.3 as fix or wait till 0.112?

**Comment by bramkragten** [comment]
@andriej 0.112

**Comment by bramkragten** [comment]
@AmirHosseinKarimi you mean HTML5 push notifications?

**Comment by AmirHosseinKarimi** [comment]
@bramkragten I think about if persist notification will receive from js-websocket and when its received front-end uses native system notification to alert the user, If connection will close after 5 minutes users will not know about notifications. Some notifications is important like attempt to login.

**Comment by balloob** [comment]
The frontend does not have a feature to use native system notifications for persistent notifications so this is not an issue. 

If you encounter problems with this behavior, please open a new issue.

**Pull request merged**